cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -569,7 +702,7 @@ class InputState {
   // returns the latest time (which is current); otherwise, returns the current time.
   // used when checking whether RHS is up to date with LHS.
   OnType GetCurrentTime() const {
-    return memo_.no_future_ ? GetLatestTime() : memo_.current_time_;
```

Comment:
Why this change?

Reply:
Because `memo_current_time_` has changed to an `std::atomic`, for which the cast extracts its value.